modules/sc-mesh-secure-deployment/src/2_0/features/cbma/tools/wpactrl.py

```diff
@@ -0,0 +1,90 @@
+import os
+import socket
+import select
+
+WPA_CTRL_MAX_REPLY_LEN = 4096
+
+class WpaCtrl:
+    counter = 0
+
+    def __init__(self, ctrl_path):
+        self.ctrl_path = ctrl_path
+        self.socket = None
+        self.local_path = f"/tmp/wpa_ctrl_{os.getpid()}-{WpaCtrl.counter}"
+        WpaCtrl.counter += 1
+
+    def __enter__(self):
+        self.socket = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM, 0)
+        self.socket.bind(self.local_path)
+        self.socket.connect(self.ctrl_path)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        os.unlink(self.local_path)
+        self.socket.close()
+
+    def request(self, cmd, msg_cb=None, reply_len=WPA_CTRL_MAX_REPLY_LEN):
+        self.socket.send(cmd)
```

Comment:
send() - don't guarantee complete packet send.

https://docs.python.org/3/howto/sockets.html#using-a-socket

You need to have known message sizes. Use sendall() to send complete packet and on receiver side you need to know how long packet is coming.